src/Stack/Upload.hs

```diff
@@ -203,13 +205,14 @@ mkUploader config us = do
         { upload_ = \tarName bytes -> do
             let formData = [partFileRequestBody "package" tarName (RequestBodyLBS bytes)]
             req2 <- formDataBody formData req1
-            let req3 = applyBasicAuth
+            mreq3 <- applyDigestAuth
                     (encodeUtf8 $ hcUsername creds)
                     (encodeUtf8 $ hcPassword creds)
                     req2
+                    manager
             putStr $ "Uploading " ++ tarName ++ "... "
             hFlush stdout
-            withResponse req3 manager $ \res ->
+            withResponse (fromMaybe req2 mreq3) manager $ \res ->
```

Comment:
As long as we handle `applyDigestAuth` failing, I'd prefer to have it logged, with sth. better than:
```haskell
when (isNull mreq3)
  $logWarn "Failure in applyDigestAuth"
```

So if a submission without password fails, there's a chance to realize that's because digest authentication failed, not because we didn't try authentication.

BTW sorry for thinking about this now, I realize multiple reviews are annoying 😅